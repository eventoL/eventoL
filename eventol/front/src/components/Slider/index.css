@import "../../../node_modules/slick-carousel/slick/slick.css";

.arrow {
    position: absolute;
    display: block;
    height: 20px;
    width: 20px;
    line-height: 0px;
    font-size: 0px;
    cursor: pointer;
    background: transparent;
    color: transparent;
    top: 50%;
    -webkit-transform: translate(0, -50%);
    -ms-transform: translate(0, -50%);
    transform: translate(0, -50%);
    padding: 0;
    border: none;
    outline: none;
    &:hover, &:focus {
        outline: none;
        background: transparent;
        color: transparent;
        &:before {
            opacity: $slick-opacity-on-hover;
        }
    }
    &.slick-disabled:before {
        opacity: $slick-opacity-not-active;
    }
    &:before {
        font-family: $slick-font-family;
        font-size: 20px;
        line-height: 1;
        color: $slick-arrow-color;
        opacity: $slick-opacity-default;
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
    }
}

.arrow-prev {
  left: -25px;
}

.arrow-next {
  right: -25px;
}
