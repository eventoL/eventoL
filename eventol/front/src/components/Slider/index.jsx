import React from 'react';
import PropTypes from 'prop-types';
import ReactSlickSlider from 'react-slick';

import Arrow from '../Arrow';
import {SLIDER_BASE_SETTINGS} from '../../utils/constants';

import './index.css';


const Slider = ({children: items}) => {
  const settings = {
    ...SLIDER_BASE_SETTINGS,
    prevArrow: <Arrow type='prev' />,
    nextArrow: <Arrow type='next' />,
  };
  return (
    <ReactSlickSlider className='container' {...settings}>
      {items}
    </ReactSlickSlider>
  );
};

Slider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

Slider.defaultProps = {
  children: null,
};

export default Slider;
