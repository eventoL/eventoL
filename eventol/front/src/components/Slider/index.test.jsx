import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('react-slick', () => 'ReactSlickSlider');
jest.mock('../Arrow', () => 'Arrow');
jest.mock('../../utils/constants', () => ({
  SLIDER_BASE_SETTINGS: {
    dots: true,
    speed: 500,
  },
}));

import Slider from '.';

describe('Slider', () => {
  let component;
  let tree;

  const getComponent = () => {
    component = renderer.create(<Slider />);
    return component;
  };

  beforeEach(() => {
    component = getComponent();
    tree = component.toJSON();
  });

  describe('Default', () => {
    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
