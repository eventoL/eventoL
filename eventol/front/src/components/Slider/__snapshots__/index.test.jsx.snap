// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Default Snapshot 1`] = `
<ReactSlickSlider
  className="container"
  dots={true}
  nextArrow={
    <Arrow
      type="next"
    />
  }
  prevArrow={
    <Arrow
      type="prev"
    />
  }
  speed={500}
/>
`;
